snow/validators/gvalidators/validator_state_server.go

```diff
@@ -79,3 +79,41 @@ func (s *Server) GetValidatorSet(ctx context.Context, req *pb.GetValidatorSetReq
 	}
 	return resp, nil
 }
+
+func (s *Server) GetCurrentValidatorSet(ctx context.Context, req *pb.GetCurrentValidatorSetRequest) (*pb.GetCurrentValidatorSetResponse, error) {
+	subnetID, err := ids.ToID(req.SubnetId)
+	if err != nil {
+		return nil, err
+	}
+
+	vdrs, currentHeight, err := s.state.GetCurrentValidatorSet(ctx, subnetID)
+	if err != nil {
+		return nil, err
+	}
+
+	resp := &pb.GetCurrentValidatorSetResponse{
+		Validators:    make([]*pb.CurrentValidator, len(vdrs)),
+		CurrentHeight: currentHeight,
+	}
+
+	i := 0
+	for _, vdr := range vdrs {
+		vdrPB := &pb.CurrentValidator{
+			NodeId:         vdr.NodeID.Bytes(),
+			StartTime:      vdr.StartTime,
+			IsActive:       vdr.IsActive,
+			ValidationId:   vdr.ValidationID[:],
+			Weight:         vdr.Weight,
+			SetWeightNonce: vdr.SetWeightNonce,
+			IsSov:          vdr.IsSoV,
+		}
+		if vdr.PublicKey != nil {
+			// This is a performance optimization to avoid the cost of compression
```

Comment:
serializing in compressed form is cheap. It is the un-compression that is expensive.

Reply:
literally copied from L73

Reply:
Understood. 

Though I think that this comment is a bit misleading - what it's probably actually trying to say is that in order to avoid the expensive decompression in the client side, the server side is not compressing.